XCUITests/AuthenticationTest.swift

```diff
@@ -228,104 +173,98 @@ class AuthenticationTest: BaseTestCase {
-        waitforExistence(app.staticTexts["Enter passcode"])
-        typePasscode("233777")
+        userState.passcode = "222222"
+        navigator.performAction(Action.ConfirmPasscodeToChangePasscode)
+
         waitforExistence(app.staticTexts["Incorrect passcode. Try again (Attempts remaining: 1)."])
         app.buttons["Cancel"].tap()
 
-        disablePasscode("133777")
+        userState.passcode = "111111"
+        navigator.nowAt(PasscodeSettings)
+        navigator.performAction(Action.DisablePasscode)
+        waitforExistence(app.tables["AuthenticationManager.settingsTableView"].staticTexts["Turn Passcode On"])
     }
 
     func testChangedPasswordMustBeNew() {
-        enablePasscode("133777")
-        app.staticTexts["Change Passcode"].tap()
-        waitforExistence(app.staticTexts["Enter passcode"])
-        typePasscode("133777")
-        waitforExistence(app.staticTexts["Enter a new passcode"])
-        typePasscode("133777")
+        navigator.performAction(Action.SetPasscode)
+        userState.newPasscode = "111111"
+
+        navigator.performAction(Action.ChangePasscode)
         waitforExistence(app.staticTexts["New passcode must be different than existing code."])
         app.navigationBars["Change Passcode"].buttons["Cancel"].tap()
 
-        disablePasscode("133777")
+        navigator.nowAt(PasscodeSettings)
+        navigator.performAction(Action.DisablePasscode)
+        waitforExistence(app.tables["AuthenticationManager.settingsTableView"].staticTexts["Turn Passcode On"])
     }
 
     func testPasscodesMustMatchWhenCreating() {
-        navigator.goto(PasscodeSettings)
-        app.tables["AuthenticationManager.settingsTableView"].staticTexts["Turn Passcode On"].tap()
-        waitforExistence(app.staticTexts["Enter a passcode"])
-        typePasscode("133777")
+        navigator.performAction(Action.SetPasscodeTypeOnce)
         waitforExistence(app.staticTexts["Re-enter passcode"])
-        typePasscode("233777")
+
+        // Enter a passcode that does not match
+        userState.newPasscode = "333333"
+        navigator.performAction(Action.SetPasscodeTypeOnce)
         waitforExistence(app.staticTexts["Passcodes didn’t match. Try again."])
         waitforExistence(app.staticTexts["Enter a passcode"])
         app.buttons["Cancel"].tap()
     }
 
     func testPasscodeMustBeCorrectWhenRemoving() {
-        enablePasscode("133777")
+        navigator.performAction(Action.SetPasscode)
         XCTAssertTrue(app.staticTexts["Immediately"].exists)
-        app.tables["AuthenticationManager.settingsTableView"].staticTexts["Turn Passcode Off"].tap()
-        waitforExistence(app.staticTexts["Enter passcode"])
-        typePasscode("233777")
 
+        navigator.performAction(Action.DisablePasscodeTypeIncorrectPasscode)
         waitforExistence(app.staticTexts["Incorrect passcode. Try again (Attempts remaining: 2)."])
-        typePasscode("133777")
+
+        navigator.nowAt(DisablePasscodeSettings)
```

Comment:
Nit: is this needed? By default, actions should return to the source screen state.